jvector-base/src/main/java/io/github/jbellis/jvector/quantization/VectorCompressor.java

```diff
@@ -74,4 +78,32 @@ default void write(DataOutput out) throws IOException {
 
     /** the size of a compressed vector */
     int compressedVectorSize();
+
+    /**
+     * Compute the mean squared error for the given vector
+     * @param vector the vector to compute the reconstruction error for
+     * @return the reconstruction error for the given vector
```

Comment:
nit: Might be good to provide the formula in the comment to make the return value clear: 

Mean squared error (MSE) per dimension, averaged over all vectors:

MSE = (sum of squared errors over all vectors and dimensions) / (number of vectors * number of dimensions)

Mathematically:

    MSE = (1 / (N * D)) * ∑ₖ ∑ᵢ ( vₖ[i] - rₖ[i] )²

Where:
    N = number of vectors
    D = dimension of each vector
    vₖ[i] = value of the i-th dimension of the k-th vector (original)
    rₖ[i] = reconstructed or quantized value of the i-th dimension (approximation)
-->